### PR TITLE
internal/dag: Add tests for Gateway status

### LINKED ISSUE #3867: Add status tests for Gateway
Add status tests for Gateway conditions that are set from the GatewayAPI Processor.

_Originally posted by @sunjayBhatia in https://github.com/projectcontour/contour/pull/3858#discussion_r663194086_

### PR DESCRIPTION
**Pull request by stevesloka**
Adds tests for the Gatewaay status that results from a DAG build.

Fixes #3867

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

File: internal/dag/gatewayapi_processor.go [hunk 1]
```diff
@@ -81,8 +81,7 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 	}
 
 	if len(p.source.gateway.Spec.Addresses) > 0 {
-		p.Error("Spec.Addresses is unsupported")
-		errs = append(errs, field.NotSupported(path, p.source.gateway.Spec.Addresses, []string{}))
+		errs = append(errs, &field.Error{Type: field.ErrorTypeNotSupported, Field: path.String(), BadValue: p.source.gateway.Spec.Addresses, Detail: "Spec.Addresses is not supported"})
```
Comment (sunjayBhatia) [inline internal/dag/gatewayapi_processor.go:84]: nit: maybe `errs` could be renamed to `gatewayErrs` or something, just so it is a little clearer what it is used for?

**Review by sunjayBhatia — APPROVED** [review]
LGTM! one small nit on naming, not a huge deal

File: internal/dag/gatewayapi_processor.go (hunk 1 above)
Comment (stevesloka) [reply internal/dag/gatewayapi_processor.go:84]: 👍

**Review by sunjayBhatia — APPROVED** [review]

**Review by skriss — APPROVED** [review]
LGTM!

**Pull request merged**